BN #24 Chores: set continous integration and test

Niyobuzima opened the pull request:
# PR Description
Please include a brief description of the pull request you want to raise.
# Description of tasks that were expected to be completed
List all the tasks and sub-tasks that were assigned to you Make sure you check every completed task. list them in this style
- [x] Configured Testing environment with either Mocha, Chai and Sinon, or Jest and Supertest to ensure that all functionalities are tested before deployment.
- [x] GitHub Actions is configured as a continuous integration service used to build and test the app.
- [x] Configured a deployment pipeline on Cyclic or Render, or any other host
# Description of tasks that were expected to be completed and that are not yet completed
- [ ] 
- [ ] 
- [ ]
# How this may be tested?
Please describe the tests that you ran to verify your functionalities. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration.
1. run `npm i` to install all required dependencies
2. run `npm run dev`
# Screenshots (If appropriate/applicable)
Provide screenshots related to your PR in case it's appropriate and applicable
# Please check this Checklist before you submit your PR:
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented on my code, particularly in hard-to-understand areas
- [x] My code generates no warnings
- [x] There are no vulnerabilities

Lydie260 committed:
Implementation of tests and git actions

Lydie260 committed:
Implementation of tests and git actions

Niyobuzima committed:
fixed error in git-actions yml file

chore(cross-env)setup cross-env
 update script running

Niyobuzima committed:
FIX modify some components and rebase

Niyobuzima committed:
FIX modify some components and rebase

Niyobuzima committed:
# This is a combination of 4 commits.
# This is the 1st commit message:

fixed error in git-actions yml file

chore(cross-env)setup cross-env
 update script running

# This is the commit message #2:

Update src/database/config/config.js

Co-authored-by: sezerano jean chrysostome <56281095+sezeranoJchrisostome@users.noreply.github.com>
# This is the commit message #3:

#BN-37 chores: setting environment variables and added dotenv to the project
# This is the commit message #4:

Delete .semaphore directory

Niyobuzima committed:
fix database configuration and rebasing

Niyobuzima committed:
bug removed config.js in the database

Niyobuzima committed:
fix error in package json

Niyobuzima committed:
# This is a combination of 11 commits.
parent 0b6f61a2c8bc37acfeddeb7c5d6ba743423f769e
author Niyobuzima <54801429+Niyobuzima@users.noreply.github.com> 1680632736 +0200
committer Niyobuzima <54801429+Niyobuzima@users.noreply.github.com> 1680632736 +0200

fix error in package json

# This is the commit message #9:

Added a post method to change the language

# This is the commit message #11:

Moved post-method endpoint for i18n

Niyobuzima committed:
bug fixed in index.js file in the src directory

Niyobuzima committed:
bug fixed in index.js file in the src directory

Niyobuzima committed:
parent 0b6f61a2c8bc37acfeddeb7c5d6ba743423f769e
author Niyobuzima <54801429+Niyobuzima@users.noreply.github.com> 1680632736 +0200
committer Niyobuzima <54801429+Niyobuzima@users.noreply.github.com> 1680632736 +0200

fix error in package json

Niyobuzima committed:
fix removed config.js in the database

Niyobuzima committed:
fix conflicts of my branch with the develop branch

Niyobuzima committed:
fix conflicts of my branch with the develop branch

Niyobuzima committed:
fix remove files in config folder

mukunzidd commented on .github/workflows/gitActions.yml:
```diff
@@ -0,0 +1,43 @@
+name: Node.js CI
```
Name this just CI or Build

mukunzidd commented on .gitignore:
Add .DS_Store in `gitignore`

mukunzidd commented on .github/workflows/gitActions.yml:
rename this file to CI so that it makes sense. The name should be more specific.

mukunzidd commented on package.json:
```diff
@@ -3,37 +3,49 @@
   "scripts": {
-    "migrate": "npx sequelize-cli db:migrate",
-    "down": "npx sequelize-cli db:migrate:undo",
+    "start": "nodemon --exec babel-node src/index",
```
- babel-node should not run in production
- run a normal node process

Niyobuzima replied:
so this feature is already on the develop so while pulling it was also included so i can't change

mukunzidd commented on package.json:
```diff
@@ -3,37 +3,49 @@
     "swagger-jsdoc": "^6.2.8",
-    "swagger-ui-express": "^4.6.2"
+    "swagger-ui-express": "^4.6.2",
+    "yaml": "^2.2.1"
```
What exactly are you using this for?

mukunzidd commented on package.json:
```diff
@@ -3,37 +3,49 @@
   "author": "",
   "license": "ISC",
   "dependencies": {
+    "@babel/cli": "^7.21.0",
+    "@babel/core": "^7.21.3",
+    "@babel/node": "^7.20.7",
+    "@babel/plugin-transform-runtime": "^7.21.0",
+    "@babel/preset-env": "^7.20.2",
+    "@babel/runtime": "^7.21.0",
```
Why duplicate @babel packages both in dev and normal dependencies. Please fix this. [Ref](https://www.freecodecamp.org/news/setup-babel-in-nodejs/)

mukunzidd commented on package.json:
```diff
@@ -3,37 +3,49 @@
     "@babel/cli": "^7.21.0",
     "@babel/core": "^7.21.3",
     "@babel/node": "^7.20.7",
     "@babel/plugin-transform-runtime": "^7.21.0",
```
Why exactly do we need this in this repo?

Illustre13 replied:
think we should have this for a common formatting style

mukunzidd reviewed (changes requested):
- Please @Niyobuzima do not mindlessly paste your PR description. Write relevant content and add a link to the relevant user user story on Trello. 
- Update the PR title. Use the format posted on the channel today. 
- Avoid grammatical error in either the title and the description. Is it set or setup?
- Reduce all the 17 commits into one comprehensive commit. Use an interactive git rebase and squash the commits. And write the new commit message according to the accepted format. Be careful lest you lose your work.

Niyobuzima committed:
fix conflicts with DS_Store file and changed the name of the action to CI

Niyobuzima committed:
fix conflicts with DS_Store file and changed the name of the action to CI

Niyobuzima committed:
fix(server) added changes to the script

Niyobuzima committed:
parent b8f0e490c46a6eabc1820ef6d2955a3b40bf869a
author Niyobuzima <54801429+Niyobuzima@users.noreply.github.com> 1680765502 +0200
committer Niyobuzima <54801429+Niyobuzima@users.noreply.github.com> 1680765502 +0200

fix(merge) conflicts with the develop branch

Illustre13 reviewed (dismissed):
Well Done!!

Niyobuzima committed:
Merge branch 'develop' into ch-set-continous-integration-and-test-#bn24

Niyobuzima committed:
fix(codeclimate)reconfigured codeclimate to meet our team standards

Niyobuzima committed:
Merge branch 'ch-set-continous-integration-and-test-#bn24' of https://github.com/atlp-rwanda/ecommerce-app-cogito-bn into ch-set-continous-integration-and-test-#bn24

:wq

Niyobuzima committed:
fix(codeclimate)reconfigured codeclimate to meet our team standards

Niyobuzima committed:
fix(conflict)resolve conflict with develop branch

Niyobuzima committed:
fix(mocha not found)fix error in package install in testing

Niyobuzima reviewed (commented):
all changes have been made

Niyobuzima reviewed (commented):
all changes have been made

Niyobuzima committed:
fix(conflict) resolve issues with develop branch

Niyobuzima committed:
fix(conflicts) resolve conflicts with develop branch

Niyobuzima committed:
Merge branch 'develop' of https://github.com/atlp-rwanda/ecommerce-app-cogito-bn into ch-set-continous-integration-and-test-#bn24

Niyobuzima committed:
Merge branch 'develop' of https://github.com/atlp-rwanda/ecommerce-app-cogito-bn into ch-set-continous-integration-and-test-#bn24

Niyobuzima committed:
Merge branch 'ch-set-continous-integration-and-test-#bn24' of https://github.com/atlp-rwanda/ecommerce-app-cogito-bn into ch-set-continous-integration-and-test-#bn24

-Added in codes in the workflow.

Niyobuzima committed:
Merge branch 'develop' of https://github.com/atlp-rwanda/ecommerce-app-cogito-bn into ch-set-continous-integration-and-test-#bn24

Niyobuzima committed:
Merge branch 'develop' of https://github.com/atlp-rwanda/ecommerce-app-cogito-bn into ch-set-continous-integration-and-test-#bn24

Niyobuzima committed:
Merge branch 'ch-set-continous-integration-and-test-#bn24' of https://github.com/atlp-rwanda/ecommerce-app-cogito-bn into ch-set-continous-integration-and-test-#bn24

-Due to confusion with userController file in git status and giving errors

Niyobuzima committed:
Merge branch 'ch-set-continous-integration-and-test-#bn24' of https://github.com/atlp-rwanda/ecommerce-app-cogito-bn into ch-set-continous-integration-and-test-#bn24

-Due to confusion with userController file in git status and giving errors

Niyobuzima committed:
Merge branch 'ch-set-continous-integration-and-test-#bn24' of https://github.com/atlp-rwanda/ecommerce-app-cogito-bn into ch-set-continous-integration-and-test-#bn24

Niyobuzima committed:
Merge branch 'ch-set-continous-integration-and-test-#bn24' of https://github.com/atlp-rwanda/ecommerce-app-cogito-bn into ch-set-continous-integration-and-test-#bn24

Niyobuzima committed:
Merge branch 'ch-set-continous-integration-and-test-#bn24' of https://github.com/atlp-rwanda/ecommerce-app-cogito-bn into ch-set-continous-integration-and-test-#bn24

Niyobuzima committed:
Merge branch 'ch-set-continous-integration-and-test-#bn24' of https://github.com/atlp-rwanda/ecommerce-app-cogito-bn into ch-set-continous-integration-and-test-#bn24

Niyobuzima committed:
Merge branch 'ch-set-continous-integration-and-test-#bn24' of https://github.com/atlp-rwanda/ecommerce-app-cogito-bn into ch-set-continous-integration-and-test-#bn24

Niyobuzima committed:
Merge branch 'ch-set-continous-integration-and-test-#bn24' of https://github.com/atlp-rwanda/ecommerce-app-cogito-bn into ch-set-continous-integration-and-test-#bn24

Niyobuzima committed:
Merge branch 'ch-set-continous-integration-and-test-#bn24' of https://github.com/atlp-rwanda/ecommerce-app-cogito-bn into ch-set-continous-integration-and-test-#bn24

Niyobuzima committed:
Merge branch 'ch-set-continous-integration-and-test-#bn24' of https://github.com/atlp-rwanda/ecommerce-app-cogito-bn into ch-set-continous-integration-and-test-#bn24

Niyobuzima committed:
Merge branch 'ch-set-continous-integration-and-test-#bn24' of https://github.com/atlp-rwanda/ecommerce-app-cogito-bn into ch-set-continous-integration-and-test-#bn24

Niyobuzima committed:
Merge branch 'ch-set-continous-integration-and-test-#bn24' of https://github.com/atlp-rwanda/ecommerce-app-cogito-bn into ch-set-continous-integration-and-test-#bn24

Niyobuzima committed:
Merge branch 'ch-set-continous-integration-and-test-#bn24' of https://github.com/atlp-rwanda/ecommerce-app-cogito-bn into ch-set-continous-integration-and-test-#bn24

Niyobuzima committed:
Merge branch 'ch-set-continous-integration-and-test-#bn24' of https://github.com/atlp-rwanda/ecommerce-app-cogito-bn into ch-set-continous-integration-and-test-#bn24

Niyobuzima committed:
Merge branch 'ch-set-continous-integration-and-test-#bn24' of https://github.com/atlp-rwanda/ecommerce-app-cogito-bn into ch-set-continous-integration-and-test-#bn24

Niyobuzima committed:
Merge branch 'ch-set-continous-integration-and-test-#bn24' of https://github.com/atlp-rwanda/ecommerce-app-cogito-bn into ch-set-continous-integration-and-test-#bn24

Niyobuzima committed:
Merge branch 'ch-set-continous-integration-and-test-#bn24' of https://github.com/atlp-rwanda/ecommerce-app-cogito-bn into ch-set-continous-integration-and-test-#bn24

Niyobuzima committed:
Merge branch 'ch-set-continous-integration-and-test-#bn24' of https://github.com/atlp-rwanda/ecommerce-app-cogito-bn into ch-set-continous-integration-and-test-#bn24

Niyobuzima committed:
Merge branch 'ch-set-continous-integration-and-test-#bn24' of https://github.com/atlp-rwanda/ecommerce-app-cogito-bn into ch-set-continous-integration-and-test-#bn24

Niyobuzima committed:
Merge branch 'ch-set-continous-integration-and-test-#bn24' of https://github.com/atlp-rwanda/ecommerce-app-cogito-bn into ch-set-continous-integration-and-test-#bn24

Niyobuzima committed:
Merge branch 'ch-set-continous-integration-and-test-#bn24' of https://github.com/atlp-rwanda/ecommerce-app-cogito-bn into ch-set-continous-integration-and-test-#bn24

Niyobuzima committed:
Merge branch 'ch-set-continous-integration-and-test-#bn24' of https://github.com/atlp-rwanda/ecommerce-app-cogito-bn into ch-set-continous-integration-and-test-#bn24

Niyobuzima committed:
Merge branch 'ch-set-continous-integration-and-test-#bn24' of https://github.com/atlp-rwanda/ecommerce-app-cogito-bn into ch-set-continous-integration-and-test-#bn24

Niyobuzima committed:
Merge branch 'ch-set-continous-integration-and-test-#bn24' of https://github.com/atlp-rwanda/ecommerce-app-cogito-bn into ch-set-continous-integration-and-test-#bn24

Niyobuzima committed:
Merge branch 'ch-set-continous-integration-and-test-#bn24' of https://github.com/atlp-rwanda/ecommerce-app-cogito-bn into ch-set-continous-integration-and-test-#bn24

Niyobuzima committed:
Merge branch 'ch-set-continous-integration-and-test-#bn24' of https://github.com/atlp-rwanda/ecommerce-app-cogito-bn into ch-set-continous-integration-and-test-#bn24

Niyobuzima committed:
Merge branch 'ch-set-continous-integration-and-test-#bn24' of https://github.com/atlp-rwanda/ecommerce-app-cogito-bn into ch-set-continous-integration-and-test-#bn24

Niyobuzima committed:
Merge branch 'ch-set-continous-integration-and-test-#bn24' of https://github.com/atlp-rwanda/ecommerce-app-cogito-bn into ch-set-continous-integration-and-test-#bn24

Niyobuzima committed:
Merge branch 'ch-set-continous-integration-and-test-#bn24' of https://github.com/atlp-rwanda/ecommerce-app-cogito-bn into ch-set-continous-integration-and-test-#bn24

Niyobuzima committed:
Merge branch 'ch-set-continous-integration-and-test-#bn24' of https://github.com/atlp-rwanda/ecommerce-app-cogito-bn into ch-set-continous-integration-and-test-#bn24

Niyobuzima committed:
Merge branch 'ch-set-continous-integration-and-test-#bn24' of https://github.com/atlp-rwanda/ecommerce-app-cogito-bn into ch-set-continous-integration-and-test-#bn24

Niyobuzima committed:
Merge branch 'ch-set-continous-integration-and-test-#bn24' of https://github.com/atlp-rwanda/ecommerce-app-cogito-bn into ch-set-continous-integration-and-test-#bn24

Niyobuzima committed:
Merge branch 'ch-set-continous-integration-and-test-#bn24' of https://github.com/atlp-rwanda/ecommerce-app-cogito-bn into ch-set-continous-integration-and-test-#bn24

Niyobuzima committed:
fix(issue) fixed issue in sending OTP
-Added new row in the migration for phones
-changes otp routes and documentation
[Finishes #21]

Niyobuzima committed:
fix(issue) fixed issue in sending OTP
-Added new row in the migration for phones
-changes otp routes and documentation
[Finishes #21]

Niyobuzima committed:
Merge branch 'ch-set-continous-integration-and-test-#bn24' of https://github.com/atlp-rwanda/ecommerce-app-cogito-bn into ch-set-continous-integration-and-test-#bn24

Niyobuzima committed:
Merge branch 'develop' of https://github.com/atlp-rwanda/ecommerce-app-cogito-bn into ch-set-continous-integration-and-test-#bn24

Niyobuzima committed:
Merge branch 'develop' of https://github.com/atlp-rwanda/ecommerce-app-cogito-bn into ch-set-continous-integration-and-test-#bn24

Niyobuzima committed:
Merge branch 'ch-set-continous-integration-and-test-#bn24' of https://github.com/atlp-rwanda/ecommerce-app-cogito-bn into ch-set-continous-integration-and-test-#bn24

Niyobuzima committed:
Merge branch 'ch-set-continous-integration-and-test-#bn24' of https://github.com/atlp-rwanda/ecommerce-app-cogito-bn into ch-set-continous-integration-and-test-#bn24

Niyobuzima committed:
Merge branch 'ch-set-continous-integration-and-test-#bn24' of https://github.com/atlp-rwanda/ecommerce-app-cogito-bn into ch-set-continous-integration-and-test-#bn24

Niyobuzima committed:
Merge branch 'ch-set-continous-integration-and-test-#bn24' of https://github.com/atlp-rwanda/ecommerce-app-cogito-bn into ch-set-continous-integration-and-test-#bn24

Niyobuzima committed:
Merge branch 'ch-set-continous-integration-and-test-#bn24' of https://github.com/atlp-rwanda/ecommerce-app-cogito-bn into ch-set-continous-integration-and-test-#bn24

Niyobuzima committed:
Merge branch 'ch-set-continous-integration-and-test-#bn24' of https://github.com/atlp-rwanda/ecommerce-app-cogito-bn into ch-set-continous-integration-and-test-#bn24

Niyobuzima committed:
Merge branch 'ch-set-continous-integration-and-test-#bn24' of https://github.com/atlp-rwanda/ecommerce-app-cogito-bn into ch-set-continous-integration-and-test-#bn24

Niyobuzima committed:
Merge branch 'ch-set-continous-integration-and-test-#bn24' of https://github.com/atlp-rwanda/ecommerce-app-cogito-bn into ch-set-continous-integration-and-test-#bn24

Niyobuzima committed:
Merge branch 'develop' of https://github.com/atlp-rwanda/ecommerce-app-cogito-bn into ch-set-continous-integration-and-test-#bn24

Niyobuzima committed:
Merge branch 'ch-set-continous-integration-and-test-#bn24' of https://github.com/atlp-rwanda/ecommerce-app-cogito-bn into ch-set-continous-integration-and-test-#bn24

Niyobuzima committed:
Merge branch 'ch-set-continous-integration-and-test-#bn24' of https://github.com/atlp-rwanda/ecommerce-app-cogito-bn into ch-set-continous-integration-and-test-#bn24

Niyobuzima committed:
Merge branch 'ch-set-continous-integration-and-test-#bn24' of https://github.com/atlp-rwanda/ecommerce-app-cogito-bn into ch-set-continous-integration-and-test-#bn24

Niyobuzima committed:
Merge branch 'ch-set-continous-integration-and-test-#bn24' of https://github.com/atlp-rwanda/ecommerce-app-cogito-bn into ch-set-continous-integration-and-test-#bn24

Niyobuzima committed:
Update user.js

Niyobuzima committed:
Merge branch 'ch-set-continous-integration-and-test-#bn24' of https://github.com/atlp-rwanda/ecommerce-app-cogito-bn into ch-set-continous-integration-and-test-#bn24

Niyobuzima committed:
Merge branch 'ch-set-continous-integration-and-test-#bn24' of https://github.com/atlp-rwanda/ecommerce-app-cogito-bn into ch-set-continous-integration-and-test-#bn24

Niyobuzima committed:
Merge branch 'ch-set-continous-integration-and-test-#bn24' of https://github.com/atlp-rwanda/ecommerce-app-cogito-bn into ch-set-continous-integration-and-test-#bn24

Niyobuzima committed:
Merge branch 'ch-set-continous-integration-and-test-#bn24' of https://github.com/atlp-rwanda/ecommerce-app-cogito-bn into ch-set-continous-integration-and-test-#bn24

Niyobuzima committed:
Merge branch 'ch-set-continous-integration-and-test-#bn24' of https://github.com/atlp-rwanda/ecommerce-app-cogito-bn into ch-set-continous-integration-and-test-#bn24

dushimeemma commented on .codeclimate.yml:
```diff
@@ -0,0 +1,22 @@
+  similar-code:
+    enabled: false
+  identical-code:
+    enabled: false
```
you can add an empty line at the end of all the files

dushimeemma commented on .vscode/settings.json:
```diff
@@ -0,0 +1,2 @@
+{
+}
```
add empty line as well

dushimeemma reviewed (approved):
Good job 🥇 🥇 🥇

dushimeemma reviewed (commented):
swatch the 91 commits into one commit and we are good to go

Illustre13 commented on .github/workflows/CI.yml:
```diff
@@ -0,0 +1,55 @@
+        run: npm run lint
+
+      - name: Run tests
+        run: npm test
```
Add an empty line to the end of file

Illustre13 commented on .sequelizerc:
```diff
@@ -2,7 +2,7 @@ require("@babel/register");
+  "config": path.resolve("./src/database/config/config.js"),
   "models-path": path.resolve("./src/database/models"),
   "seeders-path": path.resolve("./src/database/seeders"),
   "migrations-path": path.resolve("./src/database/migrations"),
```
Add empty line at the of file

Illustre13 reviewed (changes requested):
Add empty line at the end of file
